Eastern-Research-Group/expert-query
# Feature/158 update etl process

## Related Issues:
* [EQ-158](https://jira.epa.gov/browse/EQ-158)
* [EQ-163](https://jira.epa.gov/browse/EQ-163)

## Main Changes:
* Updated the cloud.gov etl process to load data from pgdumps stored in a private s3 bucket.
  * Instead of using a cron job to etl the database, this works based on a 30 minute poll. This poll checks if new data is available in the private s3 bucket. 
  * I left the logic for streaming national download files in the code base, but it is no longer used by the main etl process. We will still be able to use it via cloud foundary tasks though.
  * This new method of moving data into the database required some changes to our schema. Most of this was reording the columns to match the order of columns in the csv pgdump files. Some other changes are listed below:
    * `id` columns changed to `objectid` with the `objectId` alias.
    * Removed the `tmdlendpoint` column. Not available in the pgdump files.
    * Renamed `assessmentunitstate` to `assessmentunitstatus` for the `assessment_units` profile. In the pgdump file the column is `assessmentunitstatus`, but in the ordspub webservices it is still `assessmentunitstate`.
* Updated the sync glossary cron schedule to run every day at 1am.
* Removed sync glossary code from the main etl job. The sync glossary is already on its own cron job, so I don't think we need to run it twice on the day the database etl job is ran.
* Updated the client side to match what is in the new database.
* Updated the count in the download modal to have comma separators (i.e., 123,456 instead of 123456). 

## Steps To Test:
1. In your local database add the `s3_julian` column to the `logging.etl_schemas` table. This column will be character varying with a lenght of 20. This column should allow nulls.
2. Run the etl process
3. Verify the data is loaded.
4. Run the app
5. Verify each profile can be downloaded and verify the record count displayed in the modal has comma separators.
6. For the `assessment units` profile, verify the `assessmentUnitStatus` filter works and the `tmdlEndpoint` filter is no longer available.

## TODO Prior to Merging:
- [x] Add the `s3_julian` column to the cloud.gov dev database.
- [x] Bind the private etl s3 bucket to the etl app in cloud.gov.
- [x] Add new keys to github actions secrets.
- [x] Delete the `eq-node-s3-test` database.

## TODO After Merging:
- [ ] Test the etl process in cloud.gov.
- [ ] Delete the `eq_etl_testing` database after confirming this code works in cloud.gov. Check with Paul prior to deleting this.
- [ ] Think about how we should make the national downloads available. I think we probably should have Paul write to our public S3 bucket and then remove our private etl s3 bucket. I want to discuss this at one of our meetings to make sure Brad doesn't have any concerns about this.
